backend/ops_api/tests/ops/agreement/test_agreement.py

```diff
@@ -181,27 +181,156 @@ def test_agreement_create_grant_agreement(loaded_db):
     assert agreement.foa == "NIH"
 
 
-@pytest.mark.skip("Not yet implemented")
 @pytest.mark.usefixtures("app_ctx")
 def test_agreements_patch_by_id(auth_client, loaded_db):
+    # 400: invalid type change
+    response = auth_client.patch(
+        "/api/v1/agreements/1",
+        json={
+            "agreement_type": "GRANT",
+            "name": "Updated Grant Name",
+            "description": "Updated Grant Description",
+            "number": "AGR0001",
+        },
+    )
+    assert response.status_code == 400
+
+    # CONTRACT
     response = auth_client.patch(
         "/api/v1/agreements/1",
         json={
-            "name": "New Contract Name",
-            "description": "New Contract Description",
-            "support_contacts": [{"name": "Support Name", "email": "support@test.com"}],
+            "agreement_type": "CONTRACT",
+            "name": "Updated Grant Name",
+            "description": "Updated Grant Description",
+            "number": "AGR0001",
+            "team_members": [{"id": 1}],
+            "support_contacts": [{"id": 2}, {"id": 3}],
+            "notes": "Test Note"
+        },
+    )
+    assert response.status_code == 200
+
+    stmt = select(Agreement).where(Agreement.id == 1)
+    agreement = loaded_db.scalar(stmt)
+
+    assert agreement.name == "Updated Grant Name"
+    assert agreement.description == "Updated Grant Description"
+    assert agreement.notes == "Test Note"
+    assert [m.id for m in agreement.team_members] == [1]
+    assert [m.id for m in agreement.support_contacts] == [2, 3]
+
+    # CONTRACT - setting fields to null/empty
+    response = auth_client.patch(
+        "/api/v1/agreements/1",
+        json={
+            "agreement_type": "CONTRACT",
+            "name": "Updated Grant Name",
+            "description": "Updated Grant Description",
+            "number": "AGR0001",
+            "team_members": None,
+            "support_contacts": [],
+            "notes": None
+        },
+    )
+    assert response.status_code == 200
+
+    stmt = select(Agreement).where(Agreement.id == 1)
+    agreement = loaded_db.scalar(stmt)
+
+    assert agreement.name == "Updated Grant Name"
+    assert agreement.description == "Updated Grant Description"
+    assert agreement.notes is None
+    assert agreement.team_members == []
+    assert agreement.support_contacts == []
+
+
+@pytest.mark.usefixtures("app_ctx")
+def test_agreements_put_by_id(auth_client, loaded_db):
+    # 400: missing required agreement_type
```

Comment:
it's a matter of code style but consider splitting the test cases up into separate test functions - it can help add clarity to which test case is failing (instead of having to look at the code comments)

Reply:
done